### ISSUE TITLE
Check availability text missing

### ISSUE THREAD

**Issue by rushilg99** [macrosynergy/macrosynergy#1267]
**Describe the bug**

Most of the starting years across indicators and cross-sections are not being displayed. Yet, the underlying dataframe contains all data, and the colour coding on the plot appears correct.  
I'm using the version of the package in 'develop'.

Code:

```python
import os
import numpy as np
import pandas as pd

from tqdm import tqdm

import macrosynergy.management as msm
import macrosynergy.panel as msp
import macrosynergy.pnl as msn
import macrosynergy.signal as mss
import macrosynergy.learning as msl
from macrosynergy.download import JPMaQSDownload

from sklearn.base import BaseEstimator, RegressorMixin
from sklearn.pipeline import Pipeline
from sklearn.metrics import make_scorer
from sklearn.preprocessing import StandardScaler

import matplotlib.pyplot as plt

import warnings

warnings.simplefilter("ignore")

%load_ext nb_black

cids_dm = [
    "AUD",
    "CAD",
    "CHF",
    "EUR",
    "GBP",
    "JPY",
    "NOK",
    "NZD",
    "SEK",
    "USD",
]  # DM currency areas

cids = cids_dm

main = [
    "RYLDIRS05Y_NSA",
    "INTRGDPv5Y_NSA_P1M1ML12_3MMA",
    "CPIC_SJA_P6M6ML6AR",
    "INFTEFF_NSA",
    "PCREDITBN_SJA_P1M1ML12",
    "RGDP_SA_P1Q1QL4_20QMA",
]
econ = []
mark = ["DU05YXR_NSA"]

xcats = main + econ + mark

# Download series from J.P. Morgan DataQuery by tickers

start_date = "2000-01-01"
end_date = "2022-12-31"

tickers = [cid + "_" + xcat for cid in cids for xcat in xcats]
print(f"Maximum number of tickers is {len(tickers)}")

# Retrieve credentials

oauth_id = os.getenv("DQ_CLIENT_ID")  # Replace with own client ID
oauth_secret = os.getenv("DQ_CLIENT_SECRET")  # Replace with own secret

# Download from DataQuery

with JPMaQSDownload(client_id=oauth_id, client_secret=oauth_secret) as downloader:
    df = downloader.download(
        tickers=tickers,
        start_date=start_date,
        end_date=end_date,
        metrics=["value"],
        suppress_warning=True,
        show_progress=True,
    )

dfd = df

print("Done")

msm.check_availability(df=dfd, xcats=xcats, cids=cids, missing_recent=False)
```

**Expected behavior**

In each plot cell, a text label stating the earliest start year should be displayed.

**Screenshots**

![image](https://github.com/macrosynergy/macrosynergy/assets/67958144/d6c0dd75-a724-4d62-9575-b60822639656)
